src/bst.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals
+import random
+
+
+class Node(object):
+    """Define class for a node of a binary search tree."""
+
+    def __init__(self, val):
+        """Create an instance of Node."""
+        self.value = val
+        self.left = None
+        self.right = None
+        self.depth = 1
+
+    def find_depth(self):
+        """Find depth property of a node."""
+        if self.left is not None:
+            left_depth = self.left.depth
+        else:
+            left_depth = 0
+        if self.right is not None:
+            right_depth = self.right.depth
+        else:
+            right_depth = 0
+        self.depth = 1 + max(left_depth, right_depth)
+        return self.depth
+
+    def _insert_node(self, new_node):
+        """Insert a new node. <new_node> must be an instance of Node."""
+        parent = self
+        visited_nodes = [self]
+        while True:
+            if parent.value == new_node.value:
```

Comment:
This check isn't quite doing what you expect. Currently your tree is still inserting duplicates; per the assignment specs, you try to insert a value which is already in the tree, it should be ignored.